src/cephadm/cephadmlib/daemons/smb.py

```diff
@@ -178,20 +242,22 @@ def args(self) -> List[str]:
         return super().args() + ['init']
 
 
-class MustJoinContainer(SambaContainerCommon):
+class MustJoinContainer(SambaNetworkedInitContainer):
     def name(self) -> str:
         return 'mustjoin'
 
     def args(self) -> List[str]:
-        args = super().args() + ['must-join']
+        args = super().args()
+        if self.cfg.clustered:
+            # TODO: not only do we want to only do this on node 0, we only
+            # want to do it exactly ONCE per cluster even on pnn 0. This needs
+            # additional work to get that right.
```

Comment:
So is it like we run _must-join_ on node 0 for every attempt to expand/shrink an existing cluster(with `clustering: always`)?

Reply:
We should only run must join on the first instance of the service (rank 0, generation 0). This may need to be improved in the future to harden it against unexpected restarts etc.